newPost.sh

```diff
@@ -0,0 +1,46 @@
+read -p "Title: " title
+read -p "Date [$(date +%Y-%m-%d)]: " date
+read -p "Categories (space-separated): " categories
+read -p "Description: " description
+read -p "Post Image URL (full size): " img
+read -p "Post Image URL (smaller): " img_sm
+read -p "Author Name: " author_name
+read -p "Author Image URL: " author_img
+read -p "Author Bio: " author_bio
+read -p "Author E-Mail: " author_email
+read -p "Author GitHub URL: " author_github
+read -p "Author LinkedIn URL: " author_linkedin
+
+if [ -z "$date" ] # Default to current date
+then
+    date=$(date +%Y-%m-%d)
+fi
+
+title_url=$(echo "$title" | tr " " "-" | tr '[:upper:]' '[:lower:]')
+
+function add {
+    printf -- "$1\n" >> "_posts/$date-$title_url".md
```

Comment:
Why the use of printf instead of echo? It doesn't appear you're using any behavior specific to printf here?

I also would personally store the filename in another variable above and then check the fields related to that for confirmation before asking for other fields. That way you can ensure that the user was able to properly input information before proceeding.

Reply:
`echo` was being weird when printing white space on indented lines in the YAML, using `printf` fixed this.

Reply:
Yeah looking further down where you have the indentations that makes sense.